Assets/SEE/Utils/Paths/DataPath.cs

```diff
@@ -231,44 +204,217 @@ public string AbsolutePath
         /// system we are running on, that is, the directory separator will be \
         /// on Windows and / on all other platforms.
         /// </summary>
-        public abstract string Path { get; set; }
+        [ShowInInspector, FilePath(AbsolutePath = true)]
+        public string Path
+        {
+            get => Get();
+            set => Set(value);
+        }
 
-        protected string Get()
+        /// <summary>
+        /// Returns the absolute path.
+        ///
+        /// If this path represents a <see cref="RootKind.Url"/>, an absolute URL
+        /// is returned. The absolute URL can be either one of a foreign server or
+        /// our own data server.
+        ///
+        /// Otherwise, the returned path refers to an element in the file system.
+        ///
+        /// If this path represents a <see cref="RootKind.Absolute"/> path, that
+        /// absolute path is returned as is, that is, without replacing any directory
+        /// separator.
+        ///
+        /// Otherwise, the returned path is an absolute path as a combination of
+        /// the relative path prepended by the <see cref="GetRootPath()"/> where
+        /// a platform-dependent directory separator will be used.
+        /// </summary>
+        /// <returns>absolute path</returns>
+        private string Get()
         {
-            if (Root == RootKind.Absolute)
+            if (Root == RootKind.Url)
+            {
+                // absolutePath is set only for foreign servers, in which case relativePath
+                // will be empty. If the absolutePath is empty, the relativePath is interpreted relative
+                // to our server.
+                Uri baseUri = AbsolutePath.Length > 0 ? new(AbsolutePath) : new(Network.ClientRestAPI);
+                Uri relativeUri = new(RelativePath, UriKind.Relative);
+                return new Uri(baseUri, relativeUri).ToString();
+            }
+            else if (Root == RootKind.Absolute)
             {
-                if (string.IsNullOrEmpty(absolutePath))
+                return AbsolutePath;
+            }
+            else
+            {
+                // Path is relative to root.
+                if (string.IsNullOrEmpty(RelativePath))
```

Comment:
These if branches can be combined into the outer branches, thus flattening the nesting level here.